crates/docs/src/validation.rs

```diff
@@ -1,15 +1,46 @@
 use regex::Regex;
 use std::{
+    collections::HashMap,
     env, fs, io,
     path::{Path, PathBuf},
 };
+use walkdir::WalkDir;
 
 const EXTENSION: Option<&str> = Some("md");
 
+#[derive(Clone, Debug)]
+pub enum SnippetType {
```

Comment:
Should be more like 

```rs
pub struct SnippetType(String);
```

with convenient constructors like `SnippetType::snforge()`

docs should not be based on 2 tools that are using this crate